### PR TITLE
Run a Shell Command using Hyper

### PR DESCRIPTION
**Pull request by eIiot**
## Description

Allows the user to run a shell command in Hyper through Raycast

## Type of change

- [x] New script command

## Screenshot

CleanShot 2022-02-09 at 18 23 45

## Dependencies / Requirements

Requires [`hyperalfred`](https://github.com/gjuchault/hyperalfred) and [Hyper](https://hyper.is/)

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: commands/apps/hyper/hyper-run-shell-command.applescript [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# @raycast.schemaVersion 1
+# @raycast.title Run Shell Command
+# @raycast.mode silent
+# @raycast.packageName Hyper
+
+# Optional parameters:
+# @raycast.icon images/hyper.png
+# @raycast.currentDirectoryPath ~
+# @raycast.needsConfirmation false
+# @raycast.argument1 { "type": "text", "placeholder": "command" }
```
Comment (dehesa) [inline commands/apps/hyper/hyper-run-shell-command.applescript:19]: Please use titlecase for the placeholder (i.e. `Command`).

File: commands/apps/hyper/hyper-run-shell-command.applescript [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/osascript
+
+# Note: Hyper required
+# Install via Hyper.is: https://hyper.is/
+
+# Dependency: This script requires `hyperalfred` (https://github.com/gjuchault/hyperalfred)
+# Install via Hyper: `hyper i hyperalfred`
```
Comment (dehesa) [inline commands/apps/hyper/hyper-run-shell-command.applescript:7]: No `hyperraycast`? 😛

**Review by dehesa — CHANGES_REQUESTED** [review]
Hi @eIiot,

Nice contribution. I left a small comment, but also please reduce the icon size to 64x64 px. I would recommend you to trim some white space margin in the icon, then reduce it to 64x64 and finally pass it by a compressor, such as [ImageOptim](https://imageoptim.com/mac).

File: commands/apps/hyper/hyper-run-shell-command.applescript (hunk 2 above)
Comment (eIiot) [reply commands/apps/hyper/hyper-run-shell-command.applescript:7]: It's an adaptation of the similar Alfred script that makes compatibility possible. I could write my own, but there would be no functionality difference :)

**Comment by eIiot** [comment]
Thanks for the suggestions! I've changed the icon resolution and added the titlecase, is there anything else?

File: commands/apps/hyper/hyper-run-shell-command.applescript (hunk 2 above)
Comment (dehesa) [reply commands/apps/hyper/hyper-run-shell-command.applescript:7]: Yeah, I know I was just kidding 😄

**Review by dehesa — CHANGES_REQUESTED** [review]
Please @eIiot, follow @unnamedd advices. He is a wiser man than I am 😄

File: commands/apps/hyper/hyper-run-shell-command.applescript [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    tell application "System Events" to exists file target_file
+    if not the result then do shell script "> " & quoted form of target_file
+    set the open_target_file to open for access target_file with write permission
+    if append_data is false then set eof of the open_target_file to 0
+    write this_data to the open_target_file starting at eof
+    close access the open_target_file
+    return true
+  on error e
+    try
+      log "~/.hyper_plugins/hyperalfred.txt does not exist. Are you sure you installed hyperalfred? (https://github.com/gjuchault/hyperalfred)"
```
[the remark this answers is not in the document]
Comment (dehesa) [reply commands/apps/hyper/hyper-run-shell-command.applescript:45]: Hey @eIiot sorry for the delay,
I believe @unnamedd is referring to the content of the string (please @unnamedd correct me if I am wrong). So you could say slightly the same more tersely:
```
log "hyperalfred is required (https://github.com/gjuchault/hyperalfred)."
```

File: commands/apps/hyper/hyper-run-shell-command.applescript [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# @raycast.description Run a terminal using Hyper. Adapted from https://github.com/gjuchault/hyperalfred
+# @raycast.author Eliot Hertenstein
+# @raycast.authorURL https://github.com/eIiot
+
+on run argv
+  write_to_file( (item 1 of argv) , "./.hyper_plugins/hyperalfred.txt", false)
+  try
+    tell application "Hyper" to activate
+  on error e
+    log "Hyper is required (https://hyper.is)"
```
[the remark this answers is not in the document]
Comment (dehesa) [reply commands/apps/hyper/hyper-run-shell-command.applescript:29]: @unnamedd what do you mean exactly?

File: commands/apps/hyper/hyper-run-shell-command.applescript (hunk 3 above)
[the remark this answers is not in the document]
Comment (unnamedd) [reply commands/apps/hyper/hyper-run-shell-command.applescript:45]: You are completely right @dehesa! That's exactly what I meant here. 

Sorry for the big delay, folks!

File: commands/apps/hyper/hyper-run-shell-command.applescript (hunk 4 above)
[the remark this answers is not in the document]
Comment (unnamedd) [reply commands/apps/hyper/hyper-run-shell-command.applescript:29]: The point here it is just to follow the same pattern used up there, just to keep it conformant ;)

**Comment by eIiot** [comment]
Okay, I should have fixed all the previous issues. Anything else?

**Comment by dehesa** [comment]
Looking good, thank you @eIiot (and sorry for the back-and-forth).

**Pull request merged**